toonz/sources/toonz/scenebrowser.cpp

```diff
@@ -0,0 +1,2210 @@
+#include <QDirModel>
+#include <QDir>
+#include <QPixmap>
+#include <QUrl>
+#include <QRegExp>
+#include <QScrollBar>
+#include <QMap>
+#include <QPushButton>
+#include <QPalette>
+#include <QCheckBox>
+#include <QMutex>
+#include <QMutexLocker>
+#include <QMessageBox>
+#include <QApplication>
+#include <QFormLayout>
+#include <QMainWindow>
+#include <QLineEdit>
+#include <QTreeWidgetItem>
+#include <QSplitter>
+#include <QFileSystemWatcher>
+
+// tcg includes
+#include "tcg/boost/range_utility.h"
+#include "tcg/boost/permuted_range.h"
+
+// boost includes
+#include <boost/bind.hpp>
+#include <boost/iterator/counting_iterator.hpp>
+#include <boost/range/adaptor/filtered.hpp>
+#include <boost/range/adaptor/transformed.hpp>
+
+namespace ba = boost::adaptors;
+
+using namespace DVGui;
+
+
+//=============================================================================
+//      Local namespace
+//=============================================================================
+
+namespace {
+std::set<SceneBrowser *> activePreproductionBoards;
+//std::map<TFilePath, FCData> frameCountMap;
+//QMutex frameCountMapMutex;
+QMutex levelFileMutex;
+
+}  // namespace
+
+inline bool isMultipleFrameType(std::string type) {
+  return (type == "tlv" || type == "tzl" || type == "pli" || type == "mov" ||
+          type == "avi" || type == "3gp" || type == "gif" || type == "mp4" ||
+          type == "webm");
+}
+
+//=============================================================================
+//
+// SceneBrowserButtonBar
+//
+//-----------------------------------------------------------------------------
+
+SceneBrowserButtonBar::SceneBrowserButtonBar(DvItemViewer *itemViewer,
+                                             QWidget *parent)
+    : QToolBar(parent) {
+  setToolButtonStyle(Qt::ToolButtonTextBesideIcon);
+  setIconSize(QSize(17, 17));
+  setObjectName("buttonBar");
+  // buttonBar->setIconSize(QSize(10,10));
+
+  QIcon newFolderIcon = createQIcon("newfolder");
+  QAction *newScene  = new QAction(newFolderIcon, tr("Add new scene"), this);
+  newScene->setIconText(tr("Add scene"));
+  addAction(newScene);
+  //addSeparator();
+
+  connect(newScene, SIGNAL(triggered()), SIGNAL(newScene()));
+}
+
+//=============================================================================
+// SceneBrowser
+//-----------------------------------------------------------------------------
+
+#if QT_VERSION >= 0x050500
+SceneBrowser::SceneBrowser(QWidget *parent, Qt::WindowFlags flags,
+                         bool noContextMenu, bool multiSelectionEnabled)
+#else
+SceneBrowser::SceneBrowser(QWidget *parent, Qt::WFlags flags, bool noContextMenu,
+                         bool multiSelectionEnabled)
+#endif
+    : QFrame(parent), m_folderName(0), m_itemViewer(0) {
+  // style sheet
+  setObjectName("SceneBrowser");
+  setFrameStyle(QFrame::StyledPanel);
+
+  //m_mainSplitter      = new QSplitter(this);
+  m_folderTreeView    = new DvDirTreeView(this);
```

Comment:
Since the widget `m_folderTreeView` is not included in any layouts of this panel, it appears behind the `Add Scene` button:

![image](https://user-images.githubusercontent.com/17974955/104137067-7de57680-53dd-11eb-9068-d8242a7da6ca.png)